### PR TITLE
Support Podman

### LINKED ISSUE #196: Support Podman
Fedora linux comes with Podman pre-installed, might be nice to support it. AFAIK Podman tries to have compatible API with Docker so hopefully this will not be too much work.

Currently, I am getting this error:

```python
Connecting to docker host... 
Unable to communicate with docker on this host. This error usually
indicates that Docker is either not installed on this system, that the
docker service is not started, or that the installation is ill-
configured.  Please follow the instructions at
https://docs.docker.com/get-docker/ to install and start docker.
```

Podman has its own Python API, `podman-py`, but according to [this issue](https://github.com/containers/podman-py/issues/336), `docker-py` that we use here should work as well. I'll take a closer look.

@unkcpz in case you have any experience with Podman, I'd appreciate your thoughts.

### PR DESCRIPTION
**Pull request by danielhollas**
Podman seems to work out-of-a-box with a little extra setup. I've also tweaked the `DEFAULT_IMAGE` to include the `docker.io/` prefix since that might not be default for Podman.

Closes #196

### CODE REVIEW

**Comment by danielhollas** [comment]
I need to dig into the failed tests, apparently caused by the change in the default image.

File: aiidalab_launch/profile.py [hunk 1]
```diff
@@ -49,8 +49,8 @@ def _get_configured_host_port(container: Container) -> int | None:
     try:
         host_config = container.attrs["HostConfig"]
         return int(host_config["PortBindings"]["8888/tcp"][0]["HostPort"]) or None
-    except (KeyError, IndexError, ValueError):
-        pass
+    except (KeyError, IndexError, ValueError) as e:
```
Comment (danielhollas) [inline aiidalab_launch/profile.py:52]:
```suggestion
    except (KeyError, IndexError, ValueError):
```

File: tests/conftest.py [hunk 2]
```diff
@@ -78,7 +78,7 @@ def _pull_docker_image(docker_client):
 # Avoid interfering with used ports on the host system.
 @pytest.fixture(scope="session", autouse=True)
 def _default_port(monkeypatch_session):
-    monkeypatch_session.setattr(aiidalab_launch.profile, "DEFAULT_PORT", None)
+    monkeypatch_session.setattr(aiidalab_launch.profile, "DEFAULT_PORT", 7777)
```
Comment (danielhollas) [inline tests/conftest.py:81]: Podman seems to autoassign random port, so the tests were failing for me locally.

File: README.md [hunk 3]
```diff
@@ -162,7 +173,7 @@ SOFTWARE.
 ## Acknowledgements
 
 This work is supported by the
-[MARVEL National Centre for Competency in Research](<http://nccr-marvel.ch>) funded by the [Swiss National Science Foundation](<http://www.snf.ch/en>),
+[MARVEL National Centre for Competency in Research](<https://nccr-marvel.ch>) funded by the [Swiss National Science Foundation](<https://www.snf.ch/en>),
```
Comment (unkcpz) [inline README.md:176]: Thanks!

File: aiidalab_launch/profile.py [hunk 4]
```diff
@@ -31,7 +31,9 @@ def _default_port() -> int:  # explicit function required to enable test patchin
     return DEFAULT_PORT
 
 
-DEFAULT_IMAGE = "aiidalab/full-stack:latest"
+DEFAULT_REGISTRY = "docker.io"
```
Comment (unkcpz) [inline aiidalab_launch/profile.py:34]: docker registry has a quite strict API rate limits. What about using `ghcr.io`?

**Review by unkcpz — APPROVED** [review]
Thanks! I have no podman to test, so I approve it. 
Leave a minor open question of which registry to use.

**Pull request merged**